core/vm/contracts.libevm.go

```diff
@@ -69,6 +80,69 @@ func (p statefulPrecompile) Run([]byte) ([]byte, error) {
 	panic(fmt.Sprintf("BUG: call to %T.Run(); MUST call %T", p, p.run))
 }
 
+// A PrecompileEnvironment provides information about the context in which a
+// precompiled contract is being run.
+type PrecompileEnvironment interface {
```

Comment:
Question: in general is it preferable to define interfaces like this (no param getters) or to have a struct like:
```suggestion
type PrecompileEnvironment struct {
	Rules params.Rules
	ReadOnly bool
	StateDB StateDB
	ReadOnlyState libevm.StateReader
	Addresses *libevm.AddressContext
}
```

Reply:
Leaving this unresolved to reply to it later.